sdk/iothub/src/service/twin.rs

```diff
@@ -0,0 +1,325 @@
+use azure_core::errors::{extract_status_headers_and_body, AzureError, UnexpectedHTTPResult};
+use hyper::{Body, Client, Method, Request, StatusCode};
+use hyper_tls::HttpsConnector;
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+
+use crate::service::{ServiceClient, API_VERSION};
+
+/// AuthenticationType of a module or device
+#[derive(Debug, Deserialize)]
+pub enum AuthenticationType {
+    #[serde(rename = "certificate")]
+    Certificate,
+    #[serde(rename = "Authority")]
+    Authority,
+    #[serde(rename = "none")]
+    None,
+    #[serde(rename = "sas")]
+    SAS,
+    #[serde(rename = "selfSigned")]
+    SelfSigned,
+}
+
+/// The connection state of a module or device
+#[derive(Debug, Deserialize)]
+pub enum ConnectionState {
+    Connected,
+    Disconnected,
+}
+
+/// Device or module status
+#[derive(Deserialize, Debug)]
+pub enum Status {
```

Comment:
You can add `#[serde(rename_all = "lowercase")]` instead of handling each variant individually.